platform_plugin_teams/cms/api/views.py

```diff
@@ -0,0 +1,211 @@
+                * name (str): The name of the topic to add (required).
+                * description (str): The description of the topic to add (required).
+                * type (str): The type of the topic to add (required).
+                * max_team_size (int): The max team size of the topic to add (required).
+
+        * DELETE: /platform-plugin-teams/api/topics/{topic-id}/?course_id={course-id}
+
+            * Path Parameters:
+                * topic_id: The topic id for the topic to delete (required).
+
+            * Query Parameters:
+                * course_id: The course id for the course to delete a topic from (required).
+
+    `Example Responses`:
+
+        * POST: /platform-plugin-teams/api/topics/
+
+            * 400:
+                * The course_id is missing from the request query parameters.
+                * The topic name already exists for the course.
+
+            * 404:
+                * The supplied course_id does not exists.
+                * The supplied course is not found.
+
+            * 201: Returns a list of updated topics for the course.
+
+                The response body will contain the following fields:
+
+                * topics: A list of updated topics for the course.
+
+                    * id: The topic's unique identifier.
+                    * name: The name of the topic.
+                    * description: A description of the topic.
+                    * type: The type of the topic.
+                    * max_team_size: The max team size of the topic.
+
+        * DELETE: /platform-plugin-teams/api/topics/{topic-id}/?course_id={course-id}
+
+            * 400:
+                * The course_id is missing from the request query parameters.
+
+            * 404:
+                * The supplied course_id does not exists.
+                * The supplied course is not found.
+                * The supplied topic_id does not exists.
+
+            * 204: Returns a list of updated topics for the course.
+
+                The response body will contain the following fields:
+
+                * topics: A list of updated topics for the course.
+
+                    * id: The topic's unique identifier.
+                    * name: The name of the topic.
+                    * description: A description of the topic.
+                    * type: The type of the topic.
+                    * max_team_size: The max team size of the topic.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request):
+        """POST request handler for the topics endpoint."""
+        from cms.djangoapps.contentstore.views.course import (  # pylint: disable=import-outside-toplevel
+            update_course_advanced_settings,
+        )
```

Comment:
can we add a comment here explaining why we're doing this?

Reply:
As we are already using the CMS host for this view I would think it is no longer necessary to import this at runtime, or is it?

Reply:
We can try removing it but I think all modules are loaded at runtime so it might fail. Can you try it?

Reply:
Yes, adding it to the top also works